s3fs/core.py

```diff
@@ -752,7 +752,8 @@ async def _cat_file(self, path, version_id=None, start=None, end=None):
     async def _pipe_file(self, path, data, chunksize=50 * 2 ** 20, **kwargs):
         bucket, key, _ = self.split_path(path)
         size = len(data)
-        if size < 5 * 2 ** 20:
+        # 5 GB is the limit for an S3 PUT
+        if size < min(5 * 2 ** 30, 2 * chunksize):
```

Comment:
Why 2x the chunksize? I think it is reasonable to split an upload into 1-and-a-bit parts.
Are you thinking, because the MPU route takes a minimum of three calls?

Reply:
I used that as a heuristics to make the overhead for the multipart upload worthwhile. Also, the AWS recommendation was to use multipart upload for files starting at 100 MB that happens to be twice the default chunk size.

Reply:
OK, I'll take it :)